src/audio/up_down_mixer/up_down_mixer.c

```diff
@@ -451,5 +443,24 @@ static const struct module_interface up_down_mixer_interface = {
 	.free = up_down_mixer_free
 };
 
-DECLARE_MODULE_ADAPTER(up_down_mixer_interface, up_down_mixer_comp_uuid, up_down_mixer_comp_tr);
-SOF_MODULE_INIT(up_down_mixer, sys_comp_module_up_down_mixer_interface_init);
+DECLARE_LOADABLE_MODULE_API_VERSION(udm);
+
+static void* entry_point(void* mod_cfg, void* parent_ppl, void** mod_ptr)
+{
+	native_sys_agent = *(const struct native_system_agent**)mod_ptr;
+
+	return &up_down_mixer_interface;
+}
+
+__attribute__((section(".module")))
+const struct sof_man_module_manifest udm_manifest = {
+	.module = {
+		.name = "UPDWMIX",
+		.uuid = {0x12, 0x34, 0xf1, 0xf1, 0x12, 0x34, 0x1a, 0x34,
+			 0x8c, 0x08, 0x88, 0x4b, 0xe5, 0xd1, 0x4f, 0xaa},
+		.entry_point = (uint32_t)entry_point,
+		.type = {.load_type = SOF_MAN_MOD_TYPE_MODULE,
+		.domain_ll = 1 },
+		.affinity_mask = 1,
+	}
+};
```

Comment:
new line. Will this file now always only be built as a module? Monolithic building no longer possible?

Reply:
This is good question. Main goal is to move all build-in modules to loadable but I'm not sure if up_down_mixer at this stage will be loadable.